src/main/scala/viper/gobra/ast/frontend/Ast.scala

```diff
@@ -548,11 +551,20 @@ sealed trait PActualType extends PType
 
 sealed trait PLiteralType extends PNode
 
-sealed trait PNamedType extends PActualType {
-  def name: String
+sealed trait PTypeName extends PActualType {
+  def id : PUseLikeId
+  val name: String = id.name
 }
 
-sealed abstract class PPredeclaredType(override val name: String) extends PNamedType
+sealed trait PUnqualifiedTypeName extends PTypeName
+
+object PUnqualifiedTypeName {
+  def unapply(arg: PUnqualifiedTypeName): Option[String] = Some(arg.name)
+}
+
+sealed abstract class PPredeclaredType(override val name: String) extends PUnqualifiedTypeName with PUseLikeId {
+  override def id: PUseLikeId = this
+}
```

Comment:
could you please add a short description about when to use which and what they represent.